bundles/core/src/main/java/com/adobe/cq/commerce/core/components/internal/storefrontcontext/ProductStorefrontContextImpl.java

```diff
@@ -0,0 +1,41 @@
+/*******************************************************************************
+ *
+ *    Copyright 2021 Adobe. All rights reserved.
+ *    This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License. You may obtain a copy
+ *    of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software distributed under
+ *    the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ *    OF ANY KIND, either express or implied. See the License for the specific language
+ *    governing permissions and limitations under the License.
+ *
+ ******************************************************************************/
+package com.adobe.cq.commerce.core.components.internal.storefrontcontext;
+
+import com.adobe.cq.commerce.core.components.storefrontcontext.ProductStorefrontContext;
+import com.adobe.cq.commerce.magento.graphql.ProductInterface;
+
+public class ProductStorefrontContextImpl extends AbstractCommerceStorefrontContext implements ProductStorefrontContext {
+
+    private final ProductInterface product;
+
+    public ProductStorefrontContextImpl(ProductInterface product) {
+        this.product = product;
+    }
+
+    @Override
+    public Integer getId() {
+        return 0;
```

Comment:
? We ignore the actual product id?

Reply:
We have confirmation that internally SKU is used to match products. The id however is still a mandatory field in the schema. So as long as UID support is missing, we decided to just go with a dummy value that fulfils the schema.

Reply:
Ok maybe add a //TODO comment about this behavior and that we work around bad Events SDK design.